File: gui-cpp/src/pages/generatepage.cpp
/**
 * 页6: 生成输出 — 实现。
 */
#include "generatepage.h"
#include <QVBoxLayout>
#include <QHBoxLayout>
#include <QGroupBox>
#include <QPushButton>

GeneratePage::GeneratePage(QWidget *parent) : QWidget(parent) {
    auto *layout = new QVBoxLayout(this);
    layout->setSpacing(12);

    // 生成按钮
    auto *btnGroup = new QGroupBox("生成");
    auto *btnLayout = new QHBoxLayout(btnGroup);
    auto addBtn = [&](const char *label) -> QPushButton* {
        auto *btn = new QPushButton(label);
        btn->setMinimumWidth(140);
        btnLayout->addWidget(btn);
        return btn;
    };
    auto *allBtn = addBtn("全部生成");
    addBtn("仅 SSC xlsx");
    addBtn("仅 TwinCAT ESI");
    addBtn("仅报告");
    connect(allBtn, &QPushButton::clicked, this, &GeneratePage::generateAll);
    layout->addWidget(btnGroup);

    // 输出文件列表
    auto *outputGroup = new QGroupBox("输出文件");
    auto *outputLayout = new QVBoxLayout(outputGroup);
    m_fileList = new QListWidget;
    outputLayout->addWidget(m_fileList);
    layout->addWidget(outputGroup);

    // 状态
    m_status = new QLabel("就绪");
    layout->addWidget(m_status);
}

void GeneratePage::setStatus(const QString &text) {
    m_status->setText(text);
}

File: gui/widgets/generate_page.py
"""
Page 6: Generate Output (dev doc §12.6)

Buttons: Generate SSC xlsx, Generate TwinCAT ESI, Generate Report, Generate All
Shows: output file list after generation
"""
from PyQt5.QtWidgets import (
    QWidget, QVBoxLayout, QHBoxLayout, QPushButton,
    QGroupBox, QListWidget, QListWidgetItem, QLabel,
    QProgressBar,
)
from PyQt5.QtCore import pyqtSignal, Qt


class GeneratePage(QWidget):
    """Generate output files page."""

    # Signals to trigger generation
    generate_all = pyqtSignal()
    generate_xlsx = pyqtSignal()
    generate_esi = pyqtSignal()
    generate_report = pyqtSignal()

    def __init__(self, parent=None):
        super().__init__(parent)
        self._init_ui()

    def _init_ui(self):
        layout = QVBoxLayout(self)
        layout.setSpacing(12)

        # ── Generate buttons ───────────────────
        btn_group = QGroupBox("Generate")
        btn_layout = QHBoxLayout(btn_group)

        for text, signal in [
            ("Generate All", self.generate_all),
            ("SSC xlsx only", self.generate_xlsx),
            ("TwinCAT ESI only", self.generate_esi),
            ("Report only", self.generate_report),
        ]:
            btn = QPushButton(text)
            btn.setMinimumWidth(140)
            btn.clicked.connect(signal.emit)
            btn_layout.addWidget(btn)

        layout.addWidget(btn_group)

        # ── Progress bar ───────────────────────
        self._progress = QProgressBar()
        self._progress.setRange(0, 100)
        self._progress.setValue(0)
        layout.addWidget(self._progress)

        # ── Output file list ───────────────────
        output_group = QGroupBox("Output Files")
        output_layout = QVBoxLayout(output_group)
        self._file_list = QListWidget()
        output_layout.addWidget(self._file_list)
        layout.addWidget(output_group)

        # ── Status label ───────────────────────
        self._status = QLabel("Ready")
        layout.addWidget(self._status)

    # ── Public API ─────────────────────────────

    def set_progress(self, value: int):
        """Update progress bar (0-100)."""
        self._progress.setValue(value)

    def set_status(self, text: str):
        """Update status message."""
        self._status.setText(text)

    def set_output_files(self, files: dict[str, str]):
        """Show generated file list."""
        self._file_list.clear()
        for name, path in files.items():
            self._file_list.addItem(f"{name}: {path}")

    def append_log(self, message: str):
        """Append a log message to the status area."""
        current = self._status.text()
        self._status.setText(f"{current}\n{message}")
